File: code_samples/scala/web_development_validation.scala
/**
 * Web Development: Validation
 * AI/ML Training Sample
 */

class Validation {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object ValidationExample {
  def main(args: Array[String]): Unit = {
    val instance = new Validation()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/lua/web_development_validation.lua
-- Web Development: Validation
-- AI/ML Training Sample

Validation = {}
Validation.__ index = Validation

function Validation:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Validation:process(input)
    self.data = input
end

function Validation:getData()
    return self.data
end

function Validation:validate()
    return #self.data > 0
end

-- Example usage
local instance = Validation:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))
